packages/@uppy/companion/src/server/provider/Provider.js

```diff
@@ -64,17 +70,24 @@ class Provider {
    * @param {object} options
    * @returns {Promise}
    */
-  async deauthorizationCallback (options) { // eslint-disable-line no-unused-vars
+  // eslint-disable-next-line class-methods-use-this,no-unused-vars
+  async deauthorizationCallback (options) {
     // @todo consider doing something like throw new NotImplementedError() instead
     throw new Error('method not implemented')
   }
 
   /**
+   * Name of the OAuth provider. Return empty string if no OAuth provider is needed.
+   *
    * @returns {string}
    */
   static get authProvider () {
-    return ''
+    return noAuthProvider
   }
 }
 
 module.exports = Provider
+module.exports.noAuthProvider = noAuthProvider
+
+// OAuth providers are those that have a `static authProvider` set. It means they require OAuth authentication to work
+module.exports.isOAuthProvider = (authProvider) => authProvider !== noAuthProvider
```

Comment:
Do we need this `const noAuthProvider` as an empty string? It's not very intention revealing to me that we use an empty string. Can we use `null` instead?